src/lib/strong_typedef.hpp

```diff
@@ -16,33 +17,45 @@
-    D() BOOST_NOEXCEPT_IF(boost::has_nothrow_default_constructor<T>::value) : t() {}                              \
-    D(const D& t_) BOOST_NOEXCEPT_IF(boost::has_nothrow_copy_constructor<T>::value) : t(t_.t) {}                  \
-    D& operator=(const D& other) BOOST_NOEXCEPT_IF(boost::has_nothrow_assign<T>::value) {                         \
-      t = other.t;                                                                                                \
-      return *this;                                                                                               \
-    }                                                                                                             \
-    D& operator=(const T& other) BOOST_NOEXCEPT_IF(boost::has_nothrow_assign<T>::value) {                         \
-      t = other;                                                                                                  \
-      return *this;                                                                                               \
-    }                                                                                                             \
-    operator const T&() const { return t; }                                                                       \
-    operator T&() { return t; }                                                                                   \
-    bool operator==(const D& other) const { return t == other.t; }                                                \
-    bool operator<(const D& other) const { return t < other.t; }                                                  \
-  };                                                                                                              \
-  } /* NOLINT */                                                                                                  \
-                                                                                                                  \
-  namespace std {                                                                                                 \
-  template <>                                                                                                     \
-  struct hash<::opossum::D> : public unary_function<::opossum::D, size_t> {                                       \
-    size_t operator()(const ::opossum::D& x) const { return hash<T>{}(x); }                                       \
-  };                                                                                                              \
-  } /* NOLINT */                                                                                                  \
+#define STRONG_TYPEDEF(T, D)                                                                                        \
+  namespace opossum {                                                                                               \
```

Comment:
good job diffing this, git.

Reply:
It's because of the whitespace changes. In an earlier version, I needed more horizontal space, but I can actually go back to the old version to keep the diff small.

Reply:
Ah I see. Thats better :)